src/Service.Tests/GraphQLBuilder/MutationBuilderTests.cs

```diff
@@ -1035,17 +1036,113 @@ public void StoredProcedureEntityAsMutationField(GraphQLOperation? graphQLOperat
+                Assert.IsNotNull(field, message: FIELDNOTFOUND_ERROR);
+                string actualMutationType = field.Type.ToString();
+                Assert.AreEqual(expected: "[StoredProcedureType!]!", actual: actualMutationType, message: $"Incorrect mutation field type: {actualMutationType}");
+            }
+            catch (Exception ex)
+            {
+                if (expectsMutationField)
+                {
+                    Assert.Fail(message: $"{FIELDNOTFOUND_ERROR} {ex.Message}");
+                }
+                else
+                {
+                    Assert.IsTrue(mutationRoot.Definitions.Count == 0, message: FIELDNOTFOUND_ERROR);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Tests the GraphQL schema builder method MutationBuilder.Build()'s behavior when processing stored procedure entity configuration
+        /// which may explicitly define the field type(query/mutation) of the entity.
+        /// Only attempt to get the mutation ObjectTypeDefinitionNode when a mutation root operation type exists.
+        /// In this test, either a mutation is created or it is not, so only attempt to run GetMutationNode when a node is expected,
+        /// otherwise an exception is raised.
+        /// This is expected per GraphQL Specification (Oct 2021) https://spec.graphql.org/October2021/#sec-Root-Operation-Types
+        /// "The mutation root operation type is optional; if it is not provided, the service does not support mutations."
+        /// </summary>
+        /// <param name="graphQLOperation">Query or Mutation</param>
+        /// <param name="operations">Collection of operations denoted by their enum value, for CreateStubEntityPermissionsMap() </param>
+        /// <param name="permissionOperations">Collection of operations denoted by their string value, for GenerateStoredProcedureEntity()</param>
+        /// <param name="expectsMutationField">Whether MutationBuilder will generate a mutation field for the GraphQL schema.</param>
+        [DataTestMethod]
+        [DataRow(GraphQLOperation.Mutation, new[] { Config.Operation.Execute }, new[] { "execute" }, true, DisplayName = "Mutation field generated since all metadata is valid")]
+        [DataRow(null, new[] { Config.Operation.Execute }, new[] { "execute" }, true, DisplayName = "Mutation field generated since default operation is mutation.")]
+        [DataRow(GraphQLOperation.Mutation, new[] { Config.Operation.Read }, new[] { "read" }, false, DisplayName = "Mutation field not generated because invalid permissions were supplied")]
+        [DataRow(GraphQLOperation.Query, new[] { Config.Operation.Execute }, new[] { "execute" }, false, DisplayName = "Mutation field not generated because the configured operation is query.")]
+        public void StoredProcedureMutationFieldMetadataValidation(GraphQLOperation? graphQLOperation, Config.Operation[] operations, string[] permissionOperations, bool expectsMutationField)
```

Comment:
How is this test `StoredProcedureEntityAsMutationField` different than the test above `StoredProcedureMutationFieldMetadataValidation`?

Reply:
They weren't different. this was errata on my part. Removed.